pkg/render/node.go

```diff
@@ -231,6 +231,11 @@ func (c *nodeComponent) Objects() ([]client.Object, []client.Object) {
 		objs = append(objs, certificatemanagement.CSRClusterRole())
 	}
 
+	if c.cfg.MigrateNamespaces {
+		objs = append(objs, migration.AddClusterRoleForKubeSystemNode())
```

Comment:
To clean these up, you should add an `else` statement to this code and add the ClusterRole and ClusterRoleBinding to the list of `objsToDelete`.

Reply:
I don't think that will work in the same way that we need. The role is only needed if we are migrating from the kube-system to calico-system namespace. It's no longer needed after the migration. Adding the `else` would try to remove non-existing `ClusterRole` and `ClusterRoleBinding `if resources are already in calico-system namespace

Reply:
As I mentioned I expect that always returning the special ClusterRole and ClusterRoleBinding should be fine to do when the namespace migration is not needed (finished).